control-base/src/main/java/fi/nls/oskari/control/feature/GetWFSFeaturesHandler.java

```diff
@@ -54,44 +49,60 @@
     private static final FeatureJSON FJ = new FeatureJSON();
     private static final byte[] EMPTY_GEOJSON_FEATURE_COLLECTION = "{\"type\": \"FeatureCollection\", \"features\": []}".getBytes(StandardCharsets.UTF_8);
 
-    // Request Features in Oskari Native SRS
-    private final String requestSrsName = PropertyUtil.get(PROPERTY_NATIVE_SRS, "EPSG:4326");
-
     private PermissionHelper permissionHelper;
-    private OskariWFS110Client wfsClient;
+
+    private CoordinateReferenceSystem nativeCRS;
+    private CoordinateReferenceSystem webMercator;
 
     protected void setPermissionHelper(PermissionHelper permissionHelper) {
         this.permissionHelper = permissionHelper;
     }
 
-    protected void setWFSClient(OskariWFS110Client wfsClient) {
-        this.wfsClient = wfsClient;
-    }
-
     @Override
     public void init() {
         if (permissionHelper == null) {
             permissionHelper = new PermissionHelper(
                     ServiceFactory.getMapLayerService(),
                     ServiceFactory.getPermissionsService());
         };
-        if (wfsClient == null) {
-            wfsClient = new OskariWFS110Client();
+    }
+
+    private CoordinateReferenceSystem getNativeCRS() {
+        if (nativeCRS == null) {
+            try {
+                String nativeSrs = PropertyUtil.get(PROPERTY_NATIVE_SRS, "EPSG:4326");
+                nativeCRS = CRS.decode(nativeSrs, true);
+            } catch (Exception e) {
+                LOG.error(e, "Failed to decode Native CRS!");
+            }
         }
+        return nativeCRS;
+    }
+
+    private CoordinateReferenceSystem getWebMercator() {
+        if (webMercator == null) {
+            try {
+                webMercator = CRS.decode("EPSG:3857", true);
+            } catch (Exception e) {
+                LOG.error(e, "Failed to decode Web Mercator CRS!");
+            }
+        }
+        return webMercator;
     }
 
     @Override
     public void handleAction(ActionParameters params) throws ActionException {
         int layerId = params.getRequiredParamInt(ActionConstants.PARAM_ID);
-        Envelope bbox = parseBbox(params.getRequiredParam(PARAM_BBOX));
+        String bboxStr = params.getRequiredParam(PARAM_BBOX);
         OskariLayer layer = findLayer(layerId, params.getUser());
-        // Only support WebMercator for now
-        // String targetSrs = params.getRequiredParam(ActionConstants.PARAM_SRS);
```

Comment:
Lets keep the param option to be flexible and default to 3857 if not provided.

Reply:
Changed that in https://github.com/oskariorg/oskari-server/pull/326/commits/20a5ac386b24a8ae2e57af061a0211032cecccbb